src/main/java/util/StatsUtil.java

```diff
@@ -0,0 +1,137 @@
+package util;
+
+import javaannbench.index.Index;
+import org.apache.commons.math3.stat.descriptive.SynchronizedDescriptiveStatistics;
+import oshi.SystemInfo;
+import oshi.hardware.HardwareAbstractionLayer;
+
+import java.io.*;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class StatsUtil {
+
+    private static final String STATS_DIR = "stats/";
+    private static final String[] buildHeader = new String[]{
+            "Index", "Phases", "TotDuration", "Size",
+            "RamUsage", "AvailableMemory"
+    };
+    private static final String[] queryHeader = new String[]{
+            "Index", "TotalQueries", "AvgRecall", "Recall",
+            "AvgDuration", "TotDuration", "AvgMinFaults", "AvgMajFault", "MaxDuration",
+            "MaxMinFaults", "MaxMajFaults", "TotMinFaults", "TotMajFaults",
+            "RamUsage", "AvailableMemory"
+    };
```

Comment:
Magari aggiungi anche le configurazioni di jvector e lucene nel csv.

Ad esempio, se il csv è di lucene, ci metti come header `numCandidates`, 
mentre nell'altro ci metto `pqFactor` e `numCandidates`.

Stessa cosa nel csv di build, 
quindi metti in jvector, beamWidth, M, neighborOverflow, alpha, 
e in lucene  maxConn, beamWidth, scalarQuantization, numThreads, forceMerge nell'alltro

Reply:
Mi pare che numCandidates in jvector sia inutilizzato, vedi un po' e in caso rimuovilo

Reply:
queste info sono contenute nella descrizione dell'indice, ho aggiunto però la variabile di configurazione k nei csv-stats delle query